app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload
+    new_status = significant_status
+    if (previous_status != new_status) # we only care if something changed
```

Comment:
It's a common thing for notification to only notify on change. In our case we have 2 "CI" that are interdependent. 

If the container image build fail, the CI will fail without a doubt. So we need a way to discard the second notification.

By sending the content of `previous_status` in the webhook, the receiver can decide to discard the notification if the transition was `failure` -> `failure`.